Check that users don't put empty strings in places where they make no sense

arturbosch opened the pull request:
Proposal to fix #502 

Changes are mostly in SimpleName as this now handles all identifiers :)

In StringLiteralExpr.setValue() was a nullCheck. I'm not sure if non-empty check would be right here as some JavadocExtractorTest-Cases fail then?

arturbosch committed:
Check that users don't put empty strings in places where they make no sense - #502

arturbosch committed:
Merge branch 'master' of github.com:javaparser/javaparser

arturbosch committed:
Revert assertNonEmpty in StringLiteralExpr as JavaDocExtractorTest fails - #502

arturbosch committed:
Assert non-empty identifier in setId + cosmetic changes - #502

arturbosch committed:
Add tests for simple name - # 502

arturbosch committed:
Remove unnecessary notEmpty checks due to setIdentifier behaviour of SimpleName - # 502

matozoid commented on javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithAnnotations.java:
```diff
@@ -66,7 +64,7 @@ default N addAnnotation(AnnotationExpr element) {
     default NormalAnnotationExpr addAnnotation(String name) {
         NormalAnnotationExpr normalAnnotationExpr = new NormalAnnotationExpr(
-                parse(name), new NodeList<>());
+                Name.parse(name), new NodeList<>());
```
Huh? :D

matozoid commented on javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithName.java:
```diff
@@ -40,7 +38,7 @@
     @SuppressWarnings("unchecked")
     default N setName(String name) {
-        return setName(parse(name));
+        return setName(Name.parse(name));
```
Do you hate static imports? :)

arturbosch replied:
Well, I found both ways to use the parse method in source code, so I decided to also use my preferred way where you immediately see where the method comes from :D

matozoid replied:
Cool then :)

matozoid commented on javaparser-core/src/main/java/com/github/javaparser/ast/body/EnumDeclaration.java:
```diff
@@ -140,7 +141,7 @@ public EnumDeclaration setImplementedTypes(NodeList<ClassOrInterfaceType> implem
 
     public EnumConstantDeclaration addEnumConstant(String name) {
-        EnumConstantDeclaration enumConstant = new EnumConstantDeclaration(assertNotNull(name));
+        EnumConstantDeclaration enumConstant = new EnumConstantDeclaration(name);
```
What's happening here?

arturbosch replied:
As most expressions, statements and declarations have a SimpleName and all string aka (name, field, identifier) are passed to a new SimpleName instance many checks were not necessary now.

matozoid replied:
Hmmm, see it as a method precondition: parameter name should not be null. It is definitely not required, but it makes the stack trace clearer: it points to this location instead of the constructor of SimpleName. The user will then have to figure out why that constructor is called at all, and deduce from that nullpointer that they shouldn't have passed in a null String.

I say leave it in, the more asserts the merrier :)

arturbosch replied:
Yeah I struggled with that decision. Double check or confuse the user... As the JIT will ignore the second check I think your right. I will fix some more places.

matozoid commented on javaparser-core/src/main/java/com/github/javaparser/utils/Utils.java:
```diff
@@ -50,6 +50,13 @@
     }
 
+    public static String assertNonEmpty(String id) {
+        if (id == null || id.isEmpty()) {
```
A generic util method shouldn't be written for id's.

arturbosch replied:
You mean the parameter name? Well i couldn't come up with a proper name ... or you mean something else? I could rename it to string or something

matozoid replied:
That and the error message - dunno, I guess "String s" and "a string was unexpectedly empty" or "This string may not be empty."

arturbosch replied:
Ok will change!

arturbosch committed:
Add headers to new source files - # 502

matozoid commented on javaparser-core/src/main/java/com/github/javaparser/ast/expr/SimpleName.java:
```diff
@@ -69,8 +69,9 @@ public final String getIdentifier() {
+        assertNonEmpty(identifier);
         notifyPropertyChange(ObservableProperty.IDENTIFIER, this.identifier, identifier);
-        this.identifier = assertNotNull(identifier);
+        this.identifier = identifier;
```
<3 more asserts <3

matozoid commented on javaparser-testing/src/test/java/com/github/javaparser/ast/expr/NameTest.java:
```diff
@@ -1,6 +1,6 @@
 /*
  * Copyright (C) 2007-2010 Júlio Vilmar Gesser.
- * Copyright (C) 2011, 2013-2016 The JavaParser Team.
+ * Copyright (C) 2011, 2013-2017 The JavaParser Team.
```
Nice find! We'll have to automate updating the headers when we have the generator stuff.

arturbosch replied:
That's a task I would like to participate in. I need that tool too :). I think also about a tutorial for this case, so we can 'kill two birds with one stone' as we are saying in germany :P

matozoid committed:
Merge branch 'master' into master

matozoid committed:
Merge branch 'master' into master

arturbosch committed:
Merge branch 'master' of github.com:javaparser/javaparser

arturbosch committed:
Add more assertions when a string should be used as an identifier - #502

arturbosch committed:
Merge github.com:arturbosch/javaparser

arturbosch committed:
Make description and block tags of javadoc be retrievable

This makes possible to analyse javadoc for missing tags like parameters in a method.
